src/Patterns/Patterns.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace AstraticBlocks\Patterns;
+
+use AstraticBlocks\Patterns\Parser;
+use WP_Block_Patterns_Registry;
+use WP_Block_Pattern_Categories_Registry;
+
+class Patterns
+{
+  private string $dir;
+  private array $patterns;
+  private array $allowedFiles;
+  private Parser $parser;
+  private const ABSL_PATTERN_MAIN_CATEGORY = 'Astratic Patterns';
+
+  public function __construct()
+  {
+    $this->dir = ASBL_PATTERNS_PATH;
+    $this->patterns = array_filter(scandir($this->dir), function ($item) {
```

Comment:
what about [glob](https://www.php.net/manual/en/function.glob.php)? did you check it?

Reply:
Yes, I've checked glob(). However doesn't found any advantages over the code I've have wrote with scandir().